txnsync/encodedgroupsmarshalers.go

```diff
@@ -0,0 +1,818 @@
+	stub.BitmaskVersion.trimBitmask(int(stub.TotalTransactionsCount))
+	stub.BitmaskThreshold.trimBitmask(int(stub.TotalTransactionsCount))
+	stub.BitmaskSubsigs.trimBitmask(int(stub.TotalTransactionsCount))
+}
+
+func (stub *txGroupsEncodingStub) deconstructLsigs(i int, txn transactions.SignedTxn) {
+	bitmaskLen := bytesNeededBitmask(int(stub.TotalTransactionsCount))
+	if txn.Lsig.Logic != nil {
+		if len(stub.BitmaskLogic) == 0 {
+			stub.BitmaskLogic = make(bitmask, bitmaskLen)
+			stub.Logic = make([][]byte, 0, int(stub.TotalTransactionsCount))
+		}
+		stub.BitmaskLogic.SetBit(i)
+		stub.Logic = append(stub.Logic, txn.Lsig.Logic)
+	}
+	if txn.Lsig.Args != nil {
+		if len(stub.BitmaskLogicArgs) == 0 {
+			stub.BitmaskLogicArgs = make(bitmask, bitmaskLen)
+			stub.LogicArgs = make([][][]byte, 0, int(stub.TotalTransactionsCount))
+		}
+		stub.BitmaskLogicArgs.SetBit(i)
+		stub.LogicArgs = append(stub.LogicArgs, txn.Lsig.Args)
+	}
+	if !txn.Lsig.Sig.MsgIsZero() {
+		if len(stub.BitmaskSig) == 0 {
+			stub.BitmaskSig = make(bitmask, bitmaskLen)
+			stub.Sig = make([]byte, 0, int(stub.TotalTransactionsCount)*len(crypto.Signature{}))
+		}
+		stub.BitmaskSig.SetBit(i)
+		stub.Sig = append(stub.Sig, txn.Lsig.Sig[:]...)
+	}
+	if txn.Lsig.Msig.Version != 0 {
+		if len(stub.BitmaskVersion) == 0 {
+			stub.BitmaskVersion = make(bitmask, bitmaskLen)
+			stub.Version = make([]byte, 0, int(stub.TotalTransactionsCount))
+		}
+		stub.BitmaskVersion.SetBit(i)
+		stub.Version = append(stub.Version, txn.Lsig.Msig.Version)
+	}
+	if txn.Lsig.Msig.Threshold != 0 {
+		if len(stub.BitmaskThreshold) == 0 {
+			stub.BitmaskThreshold = make(bitmask, bitmaskLen)
+			stub.Threshold = make([]byte, 0, int(stub.TotalTransactionsCount))
+		}
+		stub.BitmaskThreshold.SetBit(i)
+		stub.Threshold = append(stub.Threshold, txn.Lsig.Msig.Threshold)
+	}
+	if txn.Lsig.Msig.Subsigs != nil {
+		if len(stub.BitmaskSubsigs) == 0 {
+			stub.BitmaskSubsigs = make(bitmask, bitmaskLen)
+			stub.Subsigs = make([][]crypto.MultisigSubsig, 0, int(stub.TotalTransactionsCount))
+		}
+		stub.BitmaskSubsigs.SetBit(i)
+		stub.Subsigs = append(stub.Subsigs, txn.Lsig.Msig.Subsigs)
+	}
+}
+
+func (stub *txGroupsEncodingStub) finishDeconstructLsigs() {
+	stub.BitmaskLogic.trimBitmask(int(stub.TotalTransactionsCount))
+	stub.BitmaskLogicArgs.trimBitmask(int(stub.TotalTransactionsCount))
+}
+
+func (stub *txGroupsEncodingStub) deconstructTransactions(i int, txn transactions.SignedTxn) error {
+	bitmaskLen := bytesNeededBitmask(int(stub.TotalTransactionsCount))
+	txTypeByte, err := TxTypeToByte(txn.Txn.Type)
+	if err != nil {
+		return fmt.Errorf("failed to deconstructTransactions: %w", err)
+	}
+	if len(stub.BitmaskTxType) == 0 {
+		stub.BitmaskTxType = make(bitmask, bitmaskLen)
+		stub.TxType = make([]byte, 0, int(stub.TotalTransactionsCount))
+	}
+	stub.TxType = append(stub.TxType, txTypeByte)
+	stub.deconstructTxnHeader(i, txn)
+	switch txn.Txn.Type {
```

Comment:
use the `txTypeByte` here instead of the `txn.Txn.Type`, since the `txTypeByte` is known to be "validated".
I know that at this point, the `txn.Txn.Type` would also be valid here, but it makes it more clear since this variable is always in "valid" state ( or none ).